libandroid-navigation/src/main/java/com/mapbox/navigation/DefaultDirectionsSession.kt

```diff
@@ -26,7 +31,7 @@ class DefaultDirectionsSession(private val router: Router, origin: Point, waypoi
             requestRoute()
         }
 
-    override var waypoints: List<Point> = waypoints
+    override var waypoints: List<PointNavigation> = waypoints
```

Comment:
should we keep `Point` as base data structure here? or are we going to wrap every single data class? i'm asking cause `Point` is a baseline object for all mapbox classes.

Reply:
If we keep `Point` as a base class users will depends on particular implementation of Mapbox geo-json library. I thought we'd like to make our own independent interfaces and models. Didn't?

Reply:
discussed, restored